src/target/esp32/CMakeLists.txt

```diff
@@ -15,3 +15,5 @@ target_link_options(${ESP_TARGET_LIB}
     PUBLIC -T${CMAKE_CURRENT_LIST_DIR}/ld/esp32.rom.libc-funcs.ld
     PUBLIC -T${CMAKE_CURRENT_LIST_DIR}/ld/esp32.rom.libgcc.ld
 )
+
+target_link_libraries(${ESP_TARGET_LIB} PRIVATE ${ESP_COMMON_LIB})
```

Comment:
Since top-level CMakeLists.txt already has access to both ${ESP_TARGET_LIB} and ${ESP_COMMON_LIB} variables can't we link them in the top-level instead of each target's cmake file?
```
add_subdirectory(src/target/${ESP_TARGET} ${ESP_TARGET_LIB})
add_subdirectory(src/target/common ${ESP_COMMON_LIB})
target_link_libraries(${ESP_TARGET_LIB} PRIVATE ${ESP_COMMON_LIB})
```